File: Python/examples/restore_object.py
# -*- coding=utf-8

from qcloud_cos import CosConfig
from qcloud_cos import CosS3Client
from qcloud_cos import CosServiceError
from qcloud_cos import CosClientError

secret_id = 'COS_SECRETID'     # 替换为用户的secret_id
secret_key = 'COS_SECRETKEY'     # 替换为用户的secret_key
region = 'COS_REGION'    # 替换为用户的region
token = None               # 使用临时密钥需要传入Token，默认为空,可不填
config = CosConfig(Region=region, SecretId=secret_id, SecretKey=secret_key, Token=token)  # 获取配置对象
client = CosS3Client(config)

# 恢复归档对象
def restore_object():
    #.cssg-snippet-body-start:[restore-object]
    response = client.restore_object(
        Bucket='examplebucket-1250000000',
        Key='exampleobject',
        RestoreRequest={
            'Days': 100,
            'CASJobParameters': {
                'Tier': 'Standard'
            }
        }
    )
    
    #.cssg-snippet-body-end

#.cssg-methods-pragma


# 恢复归档对象
restore_object()

#.cssg-methods-pragma

File: php/examples/RestoreObject.php
<?php

use Qcloud\Cos\Client;
use Qcloud\Cos\Exception\ServiceResponseException;

class RestoreObject
{
    private $cosClient;

    private $uploadId;
    private $eTag;
    private $versionId;

    // 恢复归档对象
    protected function restoreObject() {
        $cosClient = $this->cosClient;
        //.cssg-snippet-body-start:[restore-object]
        try {
            $result = $cosClient->restoreObject(array(
                'Bucket' => 'examplebucket-1250000000', //格式：BucketName-APPID
                'Key' => 'exampleobject',
                'Days' => 10,
                'CASJobParameters' => array(
                    'Tier' =>'Expedited'
                )    
            )); 
            // 请求成功
            print_r($result);
        } catch (\Exception $e) {
            // 请求失败
            echo($e);
        }
        
        //.cssg-snippet-body-end
    }

	//.cssg-methods-pragma

    protected function init() {
        $secretId = "COS_SECRETID"; //"云 API 密钥 SecretId";
        $secretKey = "COS_SECRETKEY"; //"云 API 密钥 SecretKey";
        $region = "COS_REGION"; //设置一个默认的存储桶地域
        $this->cosClient = new Qcloud\Cos\Client(
            array(
                'region' => $region,
                'schema' => 'https', //协议头部，默认为http
                'credentials'=> array(
                    'secretId'  => $secretId ,
                    'secretKey' => $secretKey)));
    }

    public function mRestoreObject() {
        $this->init();

        // 恢复归档对象
        $this->restoreObject();

	    //.cssg-methods-pragma
    }
}
?>